internal/controllers/topology/cluster/current_state.go

```diff
@@ -181,6 +193,12 @@ func (r *Reconciler) getCurrentMachineDeploymentState(ctx context.Context, clust
 		if err != nil {
 			return nil, errors.Wrap(err, fmt.Sprintf("%s Bootstrap reference could not be retrieved", tlog.KObj{Obj: m}))
 		}
+		// check that the referenced object has the ClusterTopologyOwnedLabel label.
+		// Nb. This is to make sure that a managed topology cluster does not have a reference to an object that is not
+		// owned by the topology.
+		if !labels.IsTopologyOwned(b) {
+			return nil, fmt.Errorf("referenced BootstrapTemplate object %s is not topology owned", tlog.KObj{Obj: b})
```

Comment:
What about improving the messages so they give a little bit more context, something like

```suggestion
			return nil, fmt.Errorf("BootstrapTemplate object %s referenced from MD %s is not topology owned", ...)
```
Same for all messages

Reply:
I have added your suggestions, kindly check.